src/Microsoft.AspNet.Mvc.Razor.Host/ViewHierarchyUtility.cs

```diff
@@ -77,18 +78,31 @@ private static IEnumerable<string> GetHierarchicalPath(string relativePath, stri
                 // If the specified path is for the file hierarchy being constructed, then the first file that applies
                 // to it is in a parent directory.
                 relativePath = Path.GetDirectoryName(relativePath);
+
                 if (string.IsNullOrEmpty(relativePath))
                 {
                     return Enumerable.Empty<string>();
                 }
             }
 
+            var builder = new StringBuilder(relativePath);
+            builder.Replace('\\', '/');
+
+            if (builder.Length > 0 && builder[0] != '/')
+            {
+                builder.Insert(0, '/');
+            }
```

Comment:
I was trying to get it to look like all the other paths we generate by default - https://github.com/aspnet/Mvc/pull/3316/files#diff-91b20801581acc92340a5d67c51d172dL2. We could live with out it, but we assume all `/` rooted view paths are app rooted and not absolute paths.